File: LLM-Translations/GPT-4/avatar/Java/Go/codeforces_342_B.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	input, _ := reader.ReadString('\n')
	input = strings.TrimSpace(input)
	inputArr := strings.Split(input, " ")

	m, _ := strconv.Atoi(inputArr[1])
	s, _ := strconv.Atoi(inputArr[2])
	f, _ := strconv.Atoi(inputArr[3])

	maxT := -1
	mapData := make(map[int][]int)

	for m > 0 {
		mth, _ := reader.ReadString('\n')
		mth = strings.TrimSpace(mth)
		mthArr := strings.Split(mth, " ")

		k, _ := strconv.Atoi(mthArr[0])
		mapData[k] = []int{}
		a, _ := strconv.Atoi(mthArr[1])
		b, _ := strconv.Atoi(mthArr[2])
		mapData[k] = append(mapData[k], a)
		mapData[k] = append(mapData[k], b)

		if maxT < k {
			maxT = k
		}

		m--
	}

	var actions strings.Builder
	var M byte
	if s < f {
		M = 'R'
	} else {
		M = 'L'
	}

	d := 1
	if s >= f {
		d = -1
	}

	cur := s
	a, b := -1, -1

	for t := 1; t <= maxT; t++ {
		if data, ok := mapData[t]; ok {
			a, b = data[0], data[1]
		}

		if _, ok := mapData[t]; ok && ((cur >= a && cur <= b) || (cur+d >= a && cur+d <= b)) {
			actions.WriteByte('X')
		} else {
			actions.WriteByte(M)
			cur += d
		}

		if cur == f {
			break
		}
	}

	for cur != f {
		actions.WriteByte(M)
		cur += d
	}

	fmt.Println(actions.String())
}

// //End of Code.

File: LLM-Translations/GPT-4/avatar/Java/Python/codeforces_342_B.py
import sys
from collections import defaultdict

def main():
    m, s, f = map(int, input().split())
    max_t = -1
    intervals = defaultdict(list)

    for _ in range(m):
        k, a, b = map(int, input().split())
        intervals[k] = [a, b]
        max_t = max(max_t, k)

    actions = []
    M = 'R' if s < f else 'L'
    d = 1 if s < f else -1
    cur = s
    a = -1
    b = -1

    for t in range(1, max_t + 1):
        if t in intervals:
            a, b = intervals[t]

        if t in intervals and ((cur >= a and cur <= b) or (cur + d >= a and cur + d <= b)):
            actions.append('X')
        else:
            actions.append(M)
            cur += d

        if cur == f:
            break

    while cur != f:
        actions.append(M)
        cur += d

    print("".join(actions))

if __name__ == "__main__":
    main()


